
<launch>
	<!-- ********************ARGUMENTS******************** -->
	<!-- declare args for platform -->
	<arg name="aero/platform"			 default="0" />
	<!-- declare args to be passed in for actionlib topics -->
	<arg name="aero/velocity_command_topic" default="aero/velocity_command_topic" />
	
	<!-- declare args to be passed in for topics for receiving new PC data -->
	<arg name="aero/occupancy_topic"	 default="aero/occupancy_topic" />

	<arg name="aero/odometry"     default="aero/odometry" />
	
	<!-- declare occupancy grid arguments -->
	<arg name="occupancy/update_rate"	default="0.2"/>
	<arg name="occupancy/x_dimension"	default="200"/>
	<arg name="occupancy/y_dimension"	default="200"/>
	<arg name="occupancy/z_dimension"	default="0"/>
	<arg name="occupancy/x_origin"		default="0"/>
	<arg name="occupancy/y_origin"		default="100"/>
	<arg name="occupancy/z_origin"		default="0"/>
	<arg name="occupancy/grid_resolution"	default="0.1"/>
	
	<!-- declare speed set arguments -->
	<arg name="speed_set/number"	    default="15"/>
	<arg name="speed_set/min_speed"	default="0.01"/>
	<arg name="speed_set/max_speed"	default="1.5"/>
	
	<!-- declare tentacle arguments -->
	<arg name="tentacles/number"	default="81"/>
	<arg name="tentacles/exp_factor"	default="1.15"/>
	
	
	<!-- ********************PARAMETERS******************** -->
	<!-- declare occupancy grid parameters -->
		<param name="update_rate"	value="$(arg occupancy/update_rate)"/>
		<param name="x_dimension"	value="$(arg occupancy/x_dimension)"/>
		<param name="y_dimension"	value="$(arg occupancy/y_dimension)"/>
		<param name="z_dimension"	value="$(arg occupancy/z_dimension)"/>
		<param name="x_origin"		value="$(arg occupancy/x_origin)"/>
		<param name="y_origin"		value="$(arg occupancy/y_origin)"/>
		<param name="z_origin"		value="$(arg occupancy/z_origin)"/>
		<param name="grid_resolution"	value="$(arg occupancy/grid_resolution)"/>
	
	<!-- declare speed set parameters -->
		<param name="speed_set/number"	value="$(arg speed_set/number)"/>
		<param name="speed_set/min_speed"	value="$(arg speed_set/min_speed)"/>
		<param name="speed_set/max_speed"	value="$(arg speed_set/min_speed)"/>
	
	<!-- declare tentalce parameters -->
		<param name="number"	 value="$(arg tentacles/number)"/>
		<param name="exp_factor" value="$(arg tentacles/exp_factor)"/>
	
	
	<!-- ********************INCLUDES******************** -->
	<!-- Include Sub-Launch Files -->
	<include file="$(find aero_path_planning)/launch/aero_local_planner.launch">
		<arg name="aero/velocity_command_topic" value="$(arg aero/velocity_command_topic)" />
		<arg name="aero/occupancy_topic"	 value="$(arg aero/occupancy_topic)" />
		<arg name="aero/platform"			 value="$(arg aero/platform)"  />
		<arg name="aero/odometry"     value="$(arg aero/odometry"/>
	</include>

	<include file="$(find aero_path_planning)/launch/aero_global_planner.launch">
		<arg name="aero/occupancy_topic"	 value="$(arg aero/occupancy_topic)" />
		<arg name="aero/odometry"     value="$(arg aero/odometry"/>
	</include>
</launch>
